crates/biome_js_analyze/src/lint/nursery/no_static_element_interactions.rs

```diff
@@ -0,0 +1,353 @@
+use crate::services::aria::Aria;
+use biome_analyze::context::RuleContext;
+use biome_analyze::{declare_rule, Rule, RuleDiagnostic, RuleSource};
+use biome_console::markup;
+use biome_js_syntax::jsx_ext::AnyJsxElement;
+use biome_js_syntax::{AnyJsxAttribute, JsxAttributeList};
+use biome_rowan::AstNode;
+use rustc_hash::FxHashMap;
+use std::collections::HashMap;
+
+declare_rule! {
+    /// Enforce that non-interactive, visible elements (such as `<div>`) that have click handlers use the role attribute.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <div onClick={()=>{})}></div>;
+    /// ```
+    /// ```jsx,expect_diagnostic
+    /// <span onClick={()=>{})}></span>;
+    /// ```
+    ///
+    /// When `<a>` does not have "href" attribute, that is non-interactive.
+    /// ```jsx,expect_diagnostic
+    /// <a onClick={()=>{})}></a>
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```jsx
+    /// <div role="button" onClick={()=>{})}></div>
+    /// <span role="link" onClick={()=>{})}></span>
+    /// <a href="http://example.com" onClick={()=>{})}></a>
+    /// ```
+    ///
+    pub NoStaticElementInteractions {
+        version: "next",
+        name: "noStaticElementInteractions",
+        language: "js",
+        sources: &[RuleSource::EslintJsxA11y("no-static-element-interactions")],
+        recommended: false,
+    }
+}
+
+// These are interactions defined by eslint-plugin-jsx-a11y.
+// ref: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/974275353598e9407c76bd4a50c331a953755cee/src/rules/no-static-element-interactions.js#L33-L37
+// ref: https://github.com/jsx-eslint/jsx-ast-utils/blob/main/src/eventHandlers.js
+lazy_static::lazy_static! {
+    static ref INTERACTIVE_HANDLER_LIST: HashMap<&'static str, Vec<&'static str>> = {
+        let mut m = HashMap::new();
```

Comment:
```suggestion
    static ref INTERACTIVE_HANDLER_LIST: FxHashMap<&'static str, Vec<&'static str>> = {
        let mut m = FxHashMap::default();
```

Reply:
@unvalley 
fixed in this commit aa03e21a4b645de47cdd6e01aebcb0c124d11613